### PR TITLE
Fixed docs

### PR DESCRIPTION
**Pull request by kaunteya**
### Checklist
- No code changes

### Motivation
Updated the docs for async functions. These were missed probably because they were copied from their completion-handler based counterparts.

### CODE REVIEW

File: Sources/Purchasing/Purchases/PurchasesType.swift [hunk 1]
```diff
@@ -277,13 +277,15 @@ public protocol PurchasesType: AnyObject {
...
-     * From here ``Purchases`` will handle the purchase with `StoreKit` and call the ``PurchaseCompletedBlock``.
+     * From here ``Purchases`` will handle the purchase with `StoreKit` and return ``PurchaseResultData``.
      *
-     * - Note: You do not need to finish the transaction yourself in the completion callback, ``Purchases`` will
+     * - Note: You do not need to finish the transaction yourself after this, ``Purchases`` will
      * handle this for you.
      *
      * - Parameter product: The ``StoreProduct`` the user intends to purchase.
      *
+     * If the purchase was successful there will be a ``StoreTransaction`` and a ``CustomerInfo``.
```
Comment (kaunteya) [inline Sources/Purchasing/Purchases/PurchasesType.swift:287]: Not sure if this is true!! for async API

Comment (NachoSoto) [reply Sources/Purchasing/Purchases/PurchasesType.swift:287]: Yup that's part of the tuple 👍🏻 maybe put this and the next line about `userCancelled` inside of `- Returns:` blocks?

**Review by NachoSoto — APPROVED** [review]
Great thanks! Just a small suggestion to move those new lines inside of a new `Returns` block.

**Comment by kaunteya** [comment]
Hi. Why is CustomerInfo optional in `PurchaseCompletedBlock` but non-optional in `PurchaseResultData`?
I need this info to make the doc more clear

**Comment by NachoSoto** [comment]
The reason for that is that the completion block API returns _either_ customer info or an error. When it fails with an `error`, `CustomerInfo` is `nil`. For the `async throws` APIs this is made explicit since we can ensure that it either `throws` an error or it returns a value.

**Comment by kaunteya** [comment]
@NachoSoto I have updated the docs based on your suggestions. Please have a look

**Review by NachoSoto — APPROVED** [review]
Thanks!

**Pull request merged**